sky/cloud_stores.py

```diff
@@ -209,6 +209,69 @@ def make_sync_file_command(self, source: str, destination: str) -> str:
         return ' && '.join(all_commands)
 
 
+class MinioCloudStorage(CloudStorage):
+    """Minio Cloud Storage."""
+
+    # List of commands to install AWS CLI
+    _GET_AWSCLI = [
+        'aws --version >/dev/null 2>&1 || pip3 install awscli',
+    ]
+
+    def is_directory(self, url: str) -> bool:
+        """Returns whether minio 'url' is a directory.
+
+        In cloud object stores, a "directory" refers to a regular object whose
+        name is a prefix of other objects.
+        """
+        _minio = minio.resource('s3')
+        bucket_name, path = data_utils.split_minio_path(url)
+        bucket = _minio.Bucket(bucket_name)
+
+        num_objects = 0
+        for obj in bucket.objects.filter(Prefix=path):
+            num_objects += 1
+            if obj.key == path:
+                return False
+            # If there are more than 1 object in filter, then it is a directory
+            if num_objects == 3:
```

Comment:
Should this be `num_objects > 1`?

Thanks for the PR, I'm looking at it to learn how to add new storage options.